### PR TITLE
Add initial implementation

### PR DESCRIPTION
**Pull request by chanchiwai-ray**

### CODE REVIEW

**Review by gabrielcocenza — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    source: https://github.com/prometheus-community/smartctl_exporter.git
+    source-tag: v0.12.0
+    source-type: git
+    build-snaps:
+      - go
+    stage-packages:
+      - smartmontools
+    override-pull: |
+      craftctl default
+      TAG=$(git describe --tags --abbrev=10)
```
[the remark this answers is not in the document]
Comment (chanchiwai-ray) [reply snap/snapcraft.yaml:36]: the following line removes the "v" in the tag

[the remark this answers is not in the document]
Comment (gabrielcocenza) [reply snap/snapcraft.yaml:36]: Sure, I'm just saying that it's not necessary to add the `--abbrev=10`

File: snap/snapcraft.yaml
Comment (aieri) [inline snap/snapcraft.yaml:1]: suggestion: add keys issues, source-code

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: smartctl-exporter
```
Comment (aieri) [inline snap/snapcraft.yaml:1]: question: why the switch from _ to - ? The payload's name uses the underscore

**Review by aieri — APPROVED** [review]
lgtm, with non-blocking suggestions

File: snap/snapcraft.yaml (hunk 2 above)
Comment (chanchiwai-ray) [reply snap/snapcraft.yaml:1]: ok

**Review by Pjack — DISMISSED** [review]

**Review by samuelallan72 — DISMISSED** [review]

File: snap/snapcraft.yaml (hunk 2 above)
Comment (samuelallan72) [reply snap/snapcraft.yaml]:
```
Bad snapcraft.yaml content:
- Snap names can only use ASCII lowercase letters, numbers, and hyphens, and must have at least one letter (in field 'name')
```

Looks like we can't use underscores here.

Comment (chanchiwai-ray) [reply snap/snapcraft.yaml]: Didn't know that

Comment (samuelallan72) [reply snap/snapcraft.yaml]: Me neither until I saw the CI output... TIL :slightly_smiling_face:

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+description: |
+  This snap includes smartctl_exporter to provide smartctl statistics to Prometheus.
+grade: stable
+confinement: strict
+platforms:
+  amd64:
+  arm64:
+
+apps:
+  smartctl_exporter:
```
Comment (samuelallan72) [inline snap/snapcraft.yaml:18]: @chanchiwai-ray looks like we also need to update this as well, and probably the `parts.smartctl_exporter` name too.

**Review by samuelallan72 — APPROVED** [review]

File: snap/snapcraft.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (chanchiwai-ray) [reply snap/snapcraft.yaml]: Fixed, closing

File: snap/snapcraft.yaml (hunk 2 above)
Comment (chanchiwai-ray) [reply snap/snapcraft.yaml:1]: Reverted to original naming

File: snap/snapcraft.yaml
Comment (chanchiwai-ray) [reply snap/snapcraft.yaml:1]: added

**Review by gabrielcocenza — APPROVED** [review]

**Pull request merged**